src/stores/chatExtras.js

```diff
@@ -51,6 +55,10 @@ export const useChatExtrasStore = defineStore('chatExtras', {
 		getMessageIdToEdit: (state) => (token) => {
 			return state.messageIdToEdit[token]
 		},
+
+		getNextEvent: (state) => (token) => {
+			return state.upcomingEvents[token]?.at(0)
```

Comment:
nitpick: for simplicity. `at` is used via polyfill and not well supported by TS. Classic index-access works here, as it is not a negative index.

```suggestion
			return state.upcomingEvents[token]?.[0]
```